### ISSUE TITLE
False positive for standard:discouraged-comment-location

### ISSUE THREAD

**Issue by vanniktech** [pinterest/ktlint#2364]
The following gets flagged:

```kotlin
        start(
          currentMeditation = resumed,
          beginningSound = null, // Just leads to confusion.
        )
```

The error is placed at the start of `//`:  A comment in a 'value_argument_list' is only allowed when placed on a separate line (standard:discouraged-comment-location)

Either I don't understand the exception message or I believe this is a false positive.  The function signature looks like this:

```kotlin
private fun start(
    currentMeditation: CurrentMeditation,
    beginningSound: Sound?,
)
```

**Comment by vanniktech** [comment]
Some more mentions that I have:

```kotlin
val options = listOf(
  data?.encodedSchemeSpecificPart?.takeWhile { it != '?' }, // Such as "49.631974,8.376489?q=49.631974,8.376489".
  data?.encodedQuery?.removePrefix("q="), // Such as "q=49.631974,8.376489".
)
```

```kotlin
val COLORS = listOf(
  0xFFFF0000.color, // Red.
  0xFFFFFF00.color, // Yellow.
  0xFF00FF00.color, // Green.
  0xFF00FFFF.color, // Cyan.
  0xFF0000FF.color, // Blue.
  0xFF800080.color, // Purple.
  0xFFFF00FF.color, // Magenta.
  0xFFFFFFFF.color, // White.
  0xFF808080.color, // Gray.
  0xFF000000.color, // Black.
)
```

-------------

I tried to disable that rule `ktlint_standard_discouraged-comment-location=disabled` and then I run again into https://github.com/pinterest/ktlint/issues/2338
Then it seems like I also need to `ktlint_standard_class-signature=disabled` which is a bummer because [looking at it](https://pinterest.github.io/ktlint/latest/rules/experimental/#class-signature), I'd want it. Isn't it possible for ktlint to reuse the code while still allowing a rule to be disabled? It really seems like a flaw in the design system from the user perspective.

**Comment by vanniktech** [comment]
Side note: Why does `ktlint_standard_class-signature=disabled` work when the rule in the documentation is listed as experimental? Should not it be `ktlint_experimental_class-signature=disabled`?

**Comment by paul-dingemans** [comment]
See faq https://pinterest.github.io/ktlint/latest/faq/#why-does-ktlint-discourage-certain-comment-locations and https://pinterest.github.io/ktlint/latest/faq/#why-is-a-rule-skipped-when-i-disable-some-other-rule and #2338 

> Why does ktlint_standard_class-signature=disabled work when the rule in the documentation is listed as experimental? Should not it be ktlint_experimental_class-signature=disabled

See https://pinterest.github.io/ktlint/latest/rules/experimental/

**Comment by vanniktech** [comment]
I still don't understand why putting `ktlint_standard_class-signature=disabled` in the .editorconfig file works when the rule is stated as experimental and I'd assume that I need to put `ktlint_experimental_class-signature=disabled`.

**Comment by paul-dingemans** [comment]
> I still don't understand why putting `ktlint_standard_class-signature=disabled` in the .editorconfig file works when the rule is stated as experimental and I'd assume that I need to put `ktlint_experimental_class-signature=disabled`.

Up until ktlint `0.47.x` there were two different rulesets: `standard` and `experimental`. Moving rules from `experimental` to `standard` resulted in breaking changes as the rule identifier changed from `experimental:some-rule` to `standard:some-rule`. Only the `experimental` rule set could contain `experimental` rules. Starting from `0.48.x` rules are individually marked as `experimental`. So all rules provided by ktlint are located in the ruleset `standard`. Some rules in the `standard` rule set are marked as `experimental`.

**Comment by vanniktech** [comment]
Thanks for the explanation!
